wasi/strings.go

```diff
@@ -27,12 +27,12 @@ func newNullTerminatedStrings(maxBufSize uint32, args ...string) (*nullTerminate
 	totalBufSize := uint32(0)
 	for i, arg := range args {
 		if !utf8.ValidString(arg) {
-			return nil, fmt.Errorf("arg[%d] is not a valid UTF-8 string", i)
+			return nil, fmt.Errorf("value at %d is not a valid UTF-8 string", i)
```

Comment:
I see what you are doing, but I think it is better to change the message instead.

ex. `(maxBufSize uint32, type string, vals ...string)`

then `%s[%d]..., type, i`

when you call say "arg" or "env"

remember this is internal code!